thinc/backends/ops.py

```diff
@@ -261,31 +293,20 @@ def unflatten(self, X: Floats2d, lengths: Ints1d, pad: int = 0) -> List[Floats2d
         for length in lengths:
             length = int(length)
             if pad >= 1 and length != 0:
-                X = X[pad:]
+                X = X[pad:]  # type: ignore[assignment]
             unflat.append(X[:length])
-            X = X[length:]
+            X = X[length:]  # type: ignore[assignment]
         if pad >= 1:
-            X = X[pad:]
+            X = X[pad:]  # type: ignore[assignment]
```

Comment:
Is there any way in which we can avoid these `ignore` statements?

Reply:
It would involve somehow changing the `__getitem__` (and presumably also `__setitem__` definitions in lines 50-750 of `types.py`, but there didn't seem to be an obvious place to implement the required functionality without massively restructuring what is there, which would be risky without fully understanding it.

Reply:
We could add `cast` statements? 

```
X = cast(ArrayTXd, X[pad:])
```
etc

It's not pretty, but probably better than the `type: ignore` statements?

Reply:
Awesome, thanks!